compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -838,7 +838,7 @@ import transform.SymUtils._
 
   class MatchCaseOnlyNullWarning()(using Context)
   extends PatternMatchMsg(MatchCaseOnlyNullWarningID) {
-    def msg = em"""Only ${hl("null")} is matched. Consider using ${hl("case null =>")} instead."""
+    def msg = em"""Unreachable case. If ${hl("null")} is matched, consider using ${hl("case null =>")}."""
```

Comment:
If this case is potentially reachable by null then it's a bit misleading to say it's unreachable, I think the previous wording was fine, otherwise we could go with something like this:
```suggestion
    def msg = em"""Unreachable case except for ${hl("null")} (if this is intentional, consider writing ${hl("case null =>")} instead)."""
```